# Laboratorio 2: Limpieza de datos

# Librerias

In [1]:
'''NumPy es una librería para el cálculo numérico y el análisis de datos'''
import numpy as np
'''Pandas para el análisis y manipulación de datos'''
import pandas as pd
'''El módulo csv para leer y escribir datos tabulares en formato CSV'''
import csv

# Lectura del Dataset Criminología

In [2]:
'''Carga del dataset Criminology mediante la libreía Pandas '''
df = pd.read_csv('Criminology.csv', sep=';')
'''Lectura del encabezado del dataset Criminology'''
df

CrimeId OriginalCrimeTypeName          OffenseDate CallTime  \
0      160903280     Assault / Battery  2016-03-30T00:00:00    18:42   
1      160912272    Homeless Complaint  2016-03-31T00:00:00    15:31   
2      160912590             Susp Info  2016-03-31T00:00:00    16:49   
3      160912801                Report  2016-03-31T00:00:00    17:38   
4      160912811                   594  2016-03-31T00:00:00    17:42   
...          ...                   ...                  ...      ...   
10046  160964210      Well Being Check  2016-04-05T00:00:00    23:38   
10047  160964216          Traffic Stop  2016-04-05T00:00:00    23:43   
10048  160964227          Traffic Stop  2016-04-05T00:00:00    23:48   
10049  160964229          Passing Call  2016-04-05T00:00:00    23:48   
10050  160964249          Traffic Stop  2016-04-05T00:00:00    23:54   

              CallDateTime   Disposition                  Address  \
0      2016-03-30T18:42:00           REP  100 Block Of Chilton Av   
1      2016-03-31T15:31:00           GOA  2300 Block Of Market St   
2      2016-03-31T16:49:00           GOA  2300 Block Of Market St   
3      2016-03-31T17:38:00           GOA      500 Block Of 7th St   
4      2016-03-31T17:42:00           REP       Beale St/bryant St   
...                    ...           ...                      ...   
10046  2016-04-05T23:38:00           GOA    Hollister Av/hawes St   
10047  2016-04-05T23:43:00           ADV   California St/mason St   
10048  2016-04-05T23:48:00           CIT   Rosella Ct/onondaga Av   
10049  2016-04-05T23:48:00  Not recorded          0 Blk Herbst Rd   
10050  2016-04-05T23:54:00           ADV        Grant Av/geary St   

                City State AgencyId  Range      AddressType  
0      San Francisco    CA        1    NaN  Premise Address  
1      San Francisco    CA        1    NaN  Premise Address  
2      San Francisco    CA        1    NaN  Premise Address  
3      San Francisco    CA        1    NaN  Premise Address  
4      San Francisco    CA        1    NaN     Intersection  
...              ...   ...      ...    ...              ...  
10046  San Francisco    CA        1    NaN     Intersection  
10047  San Francisco    CA        1    NaN     Intersection  
10048  San Francisco    CA        1    NaN     Intersection  
10049  San Francisco    CA        1    NaN     Geo-Override  
10050  San Francisco    CA        1    NaN     Intersection  

[10051 rows x 12 columns]

## Información de cada columna
Se observa la información de cada columna con ello se procede a identificar los valores y el total de registro que posee cada columna. Alguna columnas como City, State y Range no tienen el valor total de registro que es 10051. 

In [3]:
'''Información de que tiene cada columna'''
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10051 entries, 0 to 10050
Data columns (total 12 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   CrimeId                10051 non-null  int64  
 1   OriginalCrimeTypeName  10051 non-null  object 
 2   OffenseDate            10051 non-null  object 
 3   CallTime               10051 non-null  object 
 4   CallDateTime           10051 non-null  object 
 5   Disposition            10051 non-null  object 
 6   Address                10051 non-null  object 
 7   City                   9730 non-null   object 
 8   State                  10048 non-null  object 
 9   AgencyId               10051 non-null  object 
 10  Range                  0 non-null      float64
 11  AddressType            10051 non-null  object 
dtypes: float64(1), int64(1), object(10)
memory usage: 942.4+ KB


### Verificación de valores NULOS
Se procede a observar los valores nulos para tenerse en cuenta al momento de empezar a realizar la limpieza

In [4]:
'''Total de valores nulos en cada columna'''
df.isnull().sum()

CrimeId                      0
OriginalCrimeTypeName        0
OffenseDate                  0
CallTime                     0
CallDateTime                 0
Disposition                  0
Address                      0
City                       321
State                        3
AgencyId                     0
Range                    10051
AddressType                  0
dtype: int64